internal/pkg/store/secretsmanager/secretsmanager.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package secretsmanager wraps AWS SecretsManager API functionality.
+package secretsmanager
+
+import (
+	// "fmt"
+
+	// "github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/identity"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/session"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
+)
+
+// SecretsManager is in charge of fetching and creating projects, environment and pipeline
+// configuration in SecretsManager.
+type SecretsManager struct {
```

Comment:
Should we move this package to be under the `aws/` directory instead of `store/`?  

Although it's used for pipeline, it can technically be used to create any secrets with secretsmanager

Reply:
@kohidave had wanted it to be under store, but I agree that putting it under `aws/` might make more sense?

Reply:
_don't listen to him_

Reply:
I won't block you on this - but I'll just explain where I'm coming from. 

1. the AWS package is pretty broad in what it could do. Our tool is an AWS tool - so almost everything could fit under it.

2. I've been trying, and maybe this is wrong, packaging by responsibility > dependency, rather than by dependency > responsibility (again, almost all of our dependencies are aws). 

3. I was trying to copy my favorite CLIs a little bit like [stripe](https://github.com/stripe/stripe-cli/tree/master/pkg) and [porter](https://github.com/deislabs/porter/tree/master/pkg) and [hugo](https://github.com/gohugoio/hugo). Hugo, for example, has a deploy/ package, which calls cloudfront. (given, porter does have an [entire extension](https://github.com/deislabs/porter-aws) which is rooted in aws - I think that's because of their extension model. Even our good friends Terraform have a responsibility > dependency model like [listers/aws](https://github.com/trussle/terraform-operator/blob/23489594d4265cd67612c3f7f965a3867f487d4f/pkg/client/listers/aws/v1/awssecretsmanagersecretversion.go) for listing secrets. 

Like I said, I won't block you from putting it in AWS if you want, but the above are why I think it should go into store :)

Reply:
Thanks for the explanation :) 

I think that makes sense, I'll try to capture in a GitHub issue the layout we'd like to go for. Right now it's not consistent as each one of these packages have more responsibility than intended :|

We can keep the secrets stuff in `store`

Reply:
Let's keep it in store for now and we can discuss/refactor later!